pkgs/shelf_proxy/lib/shelf_proxy.dart

```diff
@@ -40,7 +41,15 @@ Handler proxyHandler(Object url, {http.Client? client, String? proxyName}) {
     // TODO(nweiz): Handle TRACE requests correctly. See
     // http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.8
     final requestUrl = uri.resolve(serverRequest.url.toString());
-    final clientRequest = http.StreamedRequest(serverRequest.method, requestUrl)
+
```

Comment:
We should have a changelog entry for this. It'd also be good to reference a spec/standard about this change. I'm not familiar.

ALSO: there should be a test!

Reply:
I don't how to write test, but I can support test demo.
In debug mode, web site want fetch http://targetHost/show/upgrade.
so, I use proxy fetch http://localhost:8888/show/upgrade.
And dart server is localhost:8888.

The dart server need to fetch the truly url: http://targetHost/show/upgrade.
But serverRequest.method is OPTIONS, this method can't get data, and request fail.
The dart server need use the truly method from http-header:'Access-Control-Request-Method', is GET.

Below is the screenshot:
<img width="282" alt="image" src="https://user-images.githubusercontent.com/1514899/184281963-3315229a-f9da-400f-8626-eb2d5d12119a.png">
<img width="512" alt="image" src="https://user-images.githubusercontent.com/1514899/184282096-58481a41-e4bb-4274-acf5-cc7784b33737.png">

@kevmoo Can you help me to write the test case? Thanks

Reply:
Okay. Looking here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Request-Method – it's unclear that the correct behavior is two change the method from the original request, though.

This seems like a specific need of your scenario, not something that should be done generally for every projy usage...

Reply:
> This seems like a specific need of your scenario, not something that should be done generally for every proxy usage...

+1

This does not look to match the behavior I would expect for a proxy. I don't understand how the real server could respond correctly to an `OPTIONS` request if it does not see that request type.